arbos/block_processor.go

```diff
@@ -156,7 +157,7 @@ func ProduceBlock(
 
 	hooks := NoopSequencingHooks()
 	return ProduceBlockAdvanced(
```

Comment:
blocks will usually be created by ProduceBlocks and not by ProduceBlocksAdvanced (Advanced version is for the sequencer).
I think that the right place to store tracer config is inside ExecutionEngine (execution/gethexec/executionengine.go)
Then you could add this tracer as a parameter to ProduceBlock inside createBlockFromNextMessage.